src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/LanguageServices/FixAllSpanMappingService/AbstractFixAllSpanMappingService.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CodeFixes
+{
+    internal abstract class AbstractFixAllSpanMappingService : IFixAllSpanMappingService
+    {
+        protected abstract Task<ImmutableDictionary<Document, ImmutableArray<TextSpan>>> GetFixAllSpansIfWithinGlobalStatementAsync(
+            Document document, TextSpan diagnosticSpan, FixAllScope fixAllScope, CancellationToken cancellationToken);
+
+        public async Task<ImmutableDictionary<Document, ImmutableArray<TextSpan>>> GetFixAllSpansAsync(
+            Document document, TextSpan diagnosticSpan, FixAllScope fixAllScope, CancellationToken cancellationToken)
+        {
+            Contract.ThrowIfFalse(fixAllScope is FixAllScope.ContainingMember or FixAllScope.ContainingType);
+
+            var decl = await GetContainingMemberOrTypeDeclarationAsync(document, fixAllScope, diagnosticSpan, cancellationToken).ConfigureAwait(false);
+            if (decl == null)
+                return await GetFixAllSpansIfWithinGlobalStatementAsync(document, diagnosticSpan, fixAllScope, cancellationToken).ConfigureAwait(false);
+
+            if (fixAllScope == FixAllScope.ContainingMember)
+            {
+                return ImmutableDictionary<Document, ImmutableArray<TextSpan>>.Empty
+                    .Add(document, ImmutableArray.Create(decl.FullSpan));
+            }
+            else
+            {
+                var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var symbol = semanticModel.GetDeclaredSymbol(decl, cancellationToken);
+                if (symbol?.DeclaringSyntaxReferences.Length > 1)
+                {
+                    var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+                    var builder = PooledDictionary<Document, ArrayBuilder<TextSpan>>.GetInstance();
```

Comment:
```suggestion
                    using var = PooledDictionary<Document, ArrayBuilder<TextSpan>>.GetInstance(out var builder);
```

Reply:
PooledDictionary only supports `ToImmutableMultiDictionaryAndFree`, so I ended up using this pattern.